crates/util/Cargo.toml

```diff
@@ -14,3 +14,4 @@ humantime = "2.1.0"
 rand = "0.9.0"
 tokio = { workspace = true }
 tracing = { workspace = true }
+ctrlc = "3.4"
```

Comment:
Similar to [this comment](https://github.com/spiceai/spiceai/pull/4885/files#r1972830562)
1. I found this to work better to listen for the second Ctrl-C signal. Existing solution still reports another shutdown signal even if I use it after substantial delay like 1s. This is related to how systems signal queues are implemented, I found that the channel must be drained otherwise could still be termination signal (tried `shutdown_signal` and `tokio::signal::ctrl_c().await`)
1. Taking into account this is more of dev mode, this also allows listening only for Ctrl-C and avoid forcing shutdown on other signals.